### ISSUE TITLE
Incompatible issues of cross stage aggregate operators

### ISSUE THREAD

**Issue by PHILO-HE** [oap-project/gluten#3391]
### Description

Spark sort agg will be replaced by gluten hash agg if it is transformable. But it is possible that partial agg is offloaded, but final agg cannot due to agg's post project is not supported. As partial hash agg is not compatible with final sort agg, there will have runtime issues. ObjectHash agg should have such similar issue.

One internally proposed thought is, moving unsupported post project to an independent project operator. Thus, we can make final agg transformed and just make post project fall back.

**Comment by ulysses-you** [comment]
yes, I think a independent project operator is more clear. The explain or UI can not show the post project.

**Comment by zhouyuan** [comment]
CC @weixiuli

**Comment by lgbo-ustc** [comment]
#3260 is different, it has a unsupported transform between the two stages, and the final stage doesn't fallback

**Comment by PHILO-HE** [comment]
In fallback policy, we also need to consider this issue. For example, if sort agg in last stage is offloaded to gluten hash agg, we cannot fall back the final agg stage.
